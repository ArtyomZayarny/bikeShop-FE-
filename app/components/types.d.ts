export type ListItem = {
  id: string;
};

export interface LinkItem extends ListItem {
  href: string;
  name: string;
}

export type filterItem = {
  id: string;
  label: string;
};

export interface filterBlock {
  id: string;
  label: string;
  type: string;
  items?: (filerItem | string)[];
  maxPrice?: number;
}

export interface IProduct {
  id: string;
  title: string;
  discount: number;
  imageUrl: string;
  inStock: boolean;
  price: PriceType;
  attributes: filterBlock[];
  href: string;
}

export type PriceType = {
  newPrice: number;
  oldPrice: number;
};
