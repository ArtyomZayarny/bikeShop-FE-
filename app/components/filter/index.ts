export * from "./quick-filter-btn/quick-filter-btn";
export * from "./filter";
