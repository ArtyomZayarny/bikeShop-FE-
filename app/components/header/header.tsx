import Image from "next/image";
import Link from "next/link";
import { Container } from "../container";

type headerLinkType = {
  id: string;
  label: string;
  url: string;
};

export const Header = () => {
  const headersLinks = [
    {
      id: "1",
      label: "  Про нас",
      url: "/about-us",
    },
    {
      id: "2",
      label: "Оплата і доставка",
      url: "/payment-and-delivery",
    },
    {
      id: "3",
      label: "Контакти",
      url: "/contacts",
    },
    {
      id: "4",
      label: "Блог",
      url: "/blog",
    },
  ];

  const renderHeadersLinks = (links: headerLinkType[]) => {
    return links.map((link) => (
      <li key={link.id}>
        <Link
          className="md:text-lg lg:text-sm font-[500] text-color"
          href={link.url}
        >
          {link.label}
        </Link>
      </li>
    ));
  };

  return (
    <header id="header" className="flex justify-center">
      <Container>
        {/* Left menu */}
        <ul className="flex md:gap-7 md:mb-4 justify-between lg:gap-5 lg:mb-0">
          {renderHeadersLinks(headersLinks)}
        </ul>
        {/* Right side */}
        <div className="md:justify-between flex gap-5">
          {/* Phone */}
          <p className="flex">
            <a className="font-[500]" href={`tel:+380962071551`}>
              +38 (096) 207 15 51
            </a>
            {/* add chevron icon for dropdown */}
            <Image
              src="./svg/chevron.svg"
              width={24}
              height={24}
              alt="chevron"
            />
          </p>

          {/* Lang switcher - custom radio button */}
          <div className="flex gap-2">
            <button>
              <Image src="./svg/ua.svg" alt="ua" width={29} height={14} />
            </button>
            <button>
              <Image src="./svg/en.svg" alt="en" width={27} height={14} />
            </button>
          </div>

          {/* Icons Section  */}
          <Image src="./svg/balance.svg" alt="balance" width={24} height={24} />
          <Image src="./svg/heart.svg" alt="heart" width={24} height={24} />
          <Image src="./svg/cart.svg" alt="cart" width={24} height={24} />
          <Image src="./svg/user.svg" alt="user" width={24} height={24} />
        </div>
      </Container>
    </header>
  );
};
