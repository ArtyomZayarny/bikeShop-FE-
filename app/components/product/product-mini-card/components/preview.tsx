import Image from "next/image";

type Props = {
  imageUrl: string;
};

export const Preview = ({ imageUrl }: Props) => {
  return (
    <div className="flex justify-center">
      <Image src={imageUrl} alt="trek" width={200} height={107} />
    </div>
  );
};
