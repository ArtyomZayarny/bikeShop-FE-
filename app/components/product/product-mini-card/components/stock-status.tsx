type Props = {
  inStock: boolean;
};
export const StockStatus = ({ inStock }: Props) => {
  return (
    <div>
      <span
        className={`text-sm font-normal text-${inStock ? "green" : "disabled"}`}
      >
        {inStock ? "В наявності" : "Немає в наявності"}
      </span>
    </div>
  );
};
