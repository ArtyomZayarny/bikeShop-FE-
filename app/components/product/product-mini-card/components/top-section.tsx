import Image from "next/image";
import styles from "../../styles.module.css";

type Props = {
  discount: number;
};

export const TopSection = ({ discount }: Props) => {
  return (
    <div className="flex justify-between">
      {/* discount */}
      <div>
        <span className={`text-redAccent text-sm font-semibold`}>
          -{discount}%
        </span>
      </div>
      {/* actions buttons */}
      <div className={`${styles.actionsBtn}`}>
        <div className="flex gap-2">
          <Image
            src="./svg/heart.svg"
            alt="like-button"
            width={24}
            height={24}
          />
          <Image
            src="./svg/balance.svg"
            alt="add-to-compare-list"
            width={24}
            height={24}
          />
        </div>
      </div>
    </div>
  );
};
