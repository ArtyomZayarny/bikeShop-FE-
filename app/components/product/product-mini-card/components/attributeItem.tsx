import { DropDown } from "../../../common/dropdown";
import { Title } from "../../../common/title";
import { ColorList } from "../../../filter/color-list";
import { filterBlock } from "../../../types";

type Props = {
  attr: filterBlock;
};
export const AttributeItem = ({ attr }: Props) => {
  const renderAttributeItem = () => {
    switch (attr.type) {
      case "color":
        return (
          <>
            <ColorList colors={attr.items!} />
          </>
        );
      case "size":
        return (
          <>
            <DropDown options={attr.items!} />
          </>
        );
    }
  };
  return (
    <li className="flex mb-4 flex-col">
      <Title type={6} styles="text-sm font-normal mb-2">
        {attr.label}
      </Title>
      {renderAttributeItem()}
    </li>
  );
};
