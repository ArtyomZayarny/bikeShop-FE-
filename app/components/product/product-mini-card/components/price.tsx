import { PriceType } from "@/app/components/types";

type Props = {
  price: PriceType;
};

export const Price = ({ price }: Props) => {
  return (
    <div>
      {/* price */}
      <span className="text-xs text-regular font-semibold">
        {price.newPrice} грн
      </span>
      {/* old price */}
      <span className="text-xs ml-2 text-disabled line-through">
        {price.oldPrice} грн
      </span>
    </div>
  );
};
