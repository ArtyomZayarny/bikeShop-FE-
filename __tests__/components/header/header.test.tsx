import { render } from "@testing-library/react";
import { Header } from "../../../app/components/header";

describe("Header", () => {
  it("should render Header", () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
});
