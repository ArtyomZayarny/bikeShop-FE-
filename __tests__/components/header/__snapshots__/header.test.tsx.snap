// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render Header 1`] = `
<div>
  <header
    class="flex justify-center"
    id="header"
  >
    <div
      class="justify-between py-6 w-full 2xl:max-w-screen-2xl xl:max-w-screen-xl lg:px-10 lg:flex md:px-5 "
    >
      <ul
        class="flex md:gap-7 md:mb-4 justify-between lg:gap-5 lg:mb-0"
      >
        <li>
          <a
            class="md:text-lg lg:text-sm font-[500] text-color"
            href="/about-us"
          >
              Про нас
          </a>
        </li>
        <li>
          <a
            class="md:text-lg lg:text-sm font-[500] text-color"
            href="/payment-and-delivery"
          >
            Оплата і доставка
          </a>
        </li>
        <li>
          <a
            class="md:text-lg lg:text-sm font-[500] text-color"
            href="/contacts"
          >
            Контакти
          </a>
        </li>
        <li>
          <a
            class="md:text-lg lg:text-sm font-[500] text-color"
            href="/blog"
          >
            Блог
          </a>
        </li>
      </ul>
      <div
        class="md:justify-between flex gap-5"
      >
        <p
          class="flex"
        >
          <a
            class="font-[500]"
            href="tel:+380962071551"
          >
            +38 (096) 207 15 51
          </a>
          <img
            alt="chevron"
            data-nimg="1"
            decoding="async"
            height="24"
            loading="lazy"
            src="./svg/chevron.svg"
            style="color: transparent;"
            width="24"
          />
        </p>
        <div
          class="flex gap-2"
        >
          <button>
            <img
              alt="ua"
              data-nimg="1"
              decoding="async"
              height="14"
              loading="lazy"
              src="./svg/ua.svg"
              style="color: transparent;"
              width="29"
            />
          </button>
          <button>
            <img
              alt="en"
              data-nimg="1"
              decoding="async"
              height="14"
              loading="lazy"
              src="./svg/en.svg"
              style="color: transparent;"
              width="27"
            />
          </button>
        </div>
        <img
          alt="balance"
          data-nimg="1"
          decoding="async"
          height="24"
          loading="lazy"
          src="./svg/balance.svg"
          style="color: transparent;"
          width="24"
        />
        <img
          alt="heart"
          data-nimg="1"
          decoding="async"
          height="24"
          loading="lazy"
          src="./svg/heart.svg"
          style="color: transparent;"
          width="24"
        />
        <img
          alt="cart"
          data-nimg="1"
          decoding="async"
          height="24"
          loading="lazy"
          src="./svg/cart.svg"
          style="color: transparent;"
          width="24"
        />
        <img
          alt="user"
          data-nimg="1"
          decoding="async"
          height="24"
          loading="lazy"
          src="./svg/user.svg"
          style="color: transparent;"
          width="24"
        />
      </div>
    </div>
  </header>
</div>
`;
